spec/factories/users.rb

```diff
@@ -53,7 +53,13 @@
     end
 
     factory :school_user do
-      school
+      transient do
+        school { build(:school) }
```

Comment:
i guess this will go soon. but as there's also a `school` method i'm not sure if quirky behaviour might happen

Reply:
I had to try quite a few iterations of this before ending up with a way that worked! 
This lets the specs do:
* `build(:school_user)` to have a new school constructed automatically
* `build(:school_user, school: some_other_school_object)` to supply a particular school

etc